calico/etcddriver/driver.py

```diff
@@ -512,11 +532,18 @@ def _handle_next_watcher_event(self, resync_in_progress):
         :raises FelixWriteFailed:
         :raises ResyncRequested:
         """
+        # PLW: What happens to the watcher thread if stop() gets called? Or does
+        # it plough on happily until the process goes down?
         if self._watcher_queue is None:
             raise WatcherDied()
         while not self._stop_event.is_set():
             # To make sure we always make progress, only trigger a new resync
             # if we're not in the middle of one.
+            # PLW: This means that when we get a kick to do a resync, we stop
+            # watching for a bit, so could miss an update or two. I think we
+            # should try to close that window, even though it's quite small.
+            # PLW: We could drop a resync because we are already doing one.
+            # Should at least catch and log that case.
```

Comment:
I'm souring on the periodic resync a bit.  I'm wondering if we should downplay it and disable it by default.  It's hard to see problems that it can solve without the snapshot-based approach.  WDYT?